test/type_construction_test.rb

```diff
@@ -9944,6 +9944,28 @@ def foo: () -> (Integer | String)
     end
   end
 
+  def test_type_narrowing_union_send
+    with_checker(<<~RBS) do |checker|
+        class Object
+          def present?: () -> bool
+        end
+
+        class NilClass
+          def present?: () -> false
+        end
+      RBS
+      source = parse_ruby(<<~RUBY)
+        a = [1].first
+        a + 1 if a.present?
+      RUBY
+
+      with_standard_construction(checker, source) do |construction, typing|
+        construction.synthesize(source.node)
+        assert_no_error typing
```

Comment:
This test verifies that type narrowing of a Union type occurs by confirming that no errors are raised. This verification is somewhat implicit, but is that acceptable? If there is a better testing method, please let me know.

Reply:
This is okay, but I would recommend using `type_check_test.rb` instead.

Reply:
Moved.